### ISSUE TITLE
`build --version` command not extrapolating the NEWS file name

### ISSUE THREAD

**Issue by charris** [twisted/towncrier#417]
That broke the NumPy document builds, see https://github.com/numpy/numpy/pull/22194.

**Comment by adiroiban** [comment]
Thanks for the report.

The error details

> /home/circleci/repo/doc/source/release/{version}-notes.rst: WARNING: document isn't included in any toctree

This version is not fully ignored. `build --version` works with fixed NEWS file names

```
$ towncrier build --version 1.2.3 --yes

All OK.

$ head NEWS.rst 
``towncrier`` issues are filed on `GitHub <https://github.com/hawkowl/towncrier/issues>`_, and each ticket number here corresponds to a closed GitHub issue.

.. towncrier release notes start

towncrier 1.2.3 (2022-09-01)
============================
```


----------

I think the issue was introduced here https://github.com/twisted/towncrier/pull/391

I had no idea that towncrier was designed for storing the release notes in separate files.

There wasn't much documentation about multiple file mode.

We now have some documentation.

But it looks like we don't have enough automated tests.

----------

Happy to review a PR with the fix for this.

**Comment by adiroiban** [comment]
I have now checked #391 and I can see that it has automated test.

I see this config.

You don't want multiple files, yet each file is versined.

https://github.com/numpy/numpy/blob/551a4a040308eeef6318a3bb6503da7288f6d6bc/pyproject.toml#L10-L19


I think that you want `single_file = false`

**Comment by adiroiban** [comment]
Let me know if this is a documentation issue -

The single_file is documented here https://towncrier.readthedocs.io/en/stable/#further-options

Thanks!

**Comment by charris** [comment]
Just to help, the command that causes problems is
```
      - run:
          name: create release notes
          command: |
            . venv/bin/activate
            VERSION=$(python -c "import setup; print(setup.VERSION)")
            towncrier build --version $VERSION --yes
            ./tools/ci/test_all_newsfragments_used.py
```
and the message comes from `pyproject.toml`
```
    # Do no set this since it is hard to import numpy inside the source directory
    # the name is hardcoded. Use "--version 1.18.0" to set the version
    single_file = true
    filename = "doc/source/release/{version}-notes.rst"
    ...
```
This all works with towncrier 21.9.0.  I'm not sure what the `single_file` means, but there is one output file constructed from many input fragments. The fragments aren't versioned, they refer to the number of the PR they document, only the output file is versioned.

**Comment by charris** [comment]
Note that `filename` is a template, I suspect it is no longer used as one internally by towncrier.

**Comment by adiroiban** [comment]
> only the output file is versioned.

So if the output file is versioned it means that you keep the final release notes in multiple files.

Can you try with `single_file = false` ?

--------

This is my understanding of the docs.

if you want 'single_file = true` it means all your releases are in a single file.
Towncrier will add the latest release at the start of the same file.
For this case, the file name is used as it is, without any templating.

But please review the current docs and let us know if they make sense and correclty describe the features.

From the docs
```
If single_file is set to true or unspecified, all changes will be written to a single fixed newsfile, whose name is literally fixed as the filename option.

If single_file is set to false instead, each versioned towncrier build will generate a separate newsfile, whose name is formatted as the patten given by option filename. 
```

-------

The fact that ` single_file = true` was working before, was most probably a bug which now was fixed. :)

towncrier is still kind of an alpha version.
We now try to post-mortem write the documentation and see how towncrier should work.

Thanks

**Comment by charris** [comment]
The suggested fix works. Thanks for the help.

**Comment by jenshnielsen** [comment]
Just for the reference the meaning of single_file was incorrectly inverted before it was fixed by #391 

That meant that we used to explicitly set `single_file=True`  to get multi file behavior

**Comment by adiroiban** [comment]
Thanks for the info. I hope we now have a better doc. @mikesongming did a great help to clear this confustion.
